src/Service/CacheService.php

```diff
@@ -621,7 +621,12 @@ public function removeObject(ObjectEntity $objectEntity, bool $softDelete = fals
             return;
         }
 
-        $collection->findOneAndDelete(['_id' => $identification]);
+        $schemaReference = null;
+        if ($objectEntity->getEntity()->getReference() !== null) {
+            $schemaReference = $objectEntity->getEntity()->getReference();
+        }
+
+        $collection->findOneAndDelete(['_id' => $identification, '_schema' => $schemaReference]);
```

Comment:
I don't think this will work for Elastic, see "src/Service/Cache/ElasticSearchCollection.php" -> function findOneAndDelete()

Reply:
You can see that I added code there for this right?